tests/post_processing/test_composite_run.py

```diff
@@ -63,7 +74,16 @@ def test_build_composite_run_from_two_iu_but_second_iu_ignored():
             "draw_1": [0.8, 0.9],
         }
     )
-    population_data = {"AAA00001": 100, "AAA00002": 0}
+    population_data = IUData(
```

Comment:
:bulb: I think you may have mentioned before that you like to keep things like dummy data generation each test, but I typically like any dummy data that is similar across multiple tests to be extracted to a "generation" function, to make it easier in the future if we were to add columns/inputs the the base function.

Reply:
Apologies if I've already linked to this [why good developers write bad tests](https://mtlynch.io/good-developers-bad-tests/), I think it is so good! But the idea is to keep the reader in the test so they can see everything they need to inspect that it is correct. 

By pulling into a generator, if I need to know the values to understand the test I have to flick to the generator and remember the relevant values  (in this case, what IUs are present, they're prevalence). So I don't find the benefit of saving me a bit of time when we change this data compelling enough  (we spend more time reading code more than writing it, so generally favour making reading easier over writing). 

If there was a lot of boiler plate in the data I'd agree, but since basically everything is relevant I think it is best to keep it here.

That said - happy to discuss if you still disagree - definitely a balance to be struck with this sort of decision!

Reply:
Nope no worries, this implementation is totally fine! I definitely understand the reasoning behind it, its more of a preference on my side than anything else.